src/components/Reports/GoalCalculator/GoalCalculator.tsx

```diff
@@ -168,66 +145,78 @@ export const GoalCalculator: React.FC<GoalCalculatorProps> = ({
+                      })}
+                    >
+                      {complete ? (
                         <CircleIcon sx={{ fontSize: '1rem' }} />
                       ) : (
                         <RadioButtonUncheckedIcon sx={{ fontSize: '1rem' }} />
                       )}
-                    </StyledStepIcon>
-                  </StyledStepListItemIcon>
+                    </Box>
+                  </CategoryListItemIcon>
                   <ListItemText
                     primary={title}
                     primaryTypographyProps={{ variant: 'body2' }}
                   />
-                </StyledStepListItemButton>
+                </CategoryListItem>
               );
             })}
           </List>
         </StyledDrawer>
         {isDrawerOpen && <Divider orientation="vertical" flexItem />}
-        <Box flex={1}>
-          <StyledToolbar disableGutters>
-            <StyledTitle variant="h6">
-              {stepTitle || t('Goal Calculator')}
-            </StyledTitle>
-            {!!rightPanelComponent && (
-              <IconButton
-                onClick={toggleRightPanel}
-                aria-label={
-                  isRightOpen ? t('Hide Right Panel') : t('Show Right Panel')
-                }
-              >
-                <InfoIcon />
-              </IconButton>
-            )}
-          </StyledToolbar>
-          <Box>
-            {stepComponent || (
-              <StyledDefaultContent>
-                <Typography variant="body1">
-                  {t(
-                    'Please select a step from the left panel to view its content.',
+        <Divider orientation="vertical" flexItem />
+        <CategoriesStack flex={1} spacing={4} divider={<Divider />}>
+          {categories?.map((category) => {
+            const rightPanelContent = category.rightPanelComponent;
+            return (
+              <CategoryContainer key={category.id}>
+                <Typography variant="h6">
+                  {category.title}
+                  {rightPanelContent && (
+                    <IconButton
+                      onClick={() => {
+                        setRightPanelContent(rightPanelContent);
+                      }}
+                      aria-label={t('Show additional info')}
+                    >
+                      <InfoIcon />
+                    </IconButton>
                   )}
                 </Typography>
-              </StyledDefaultContent>
-            )}
-          </Box>
-        </Box>
+                {category.component}
+              </CategoryContainer>
+            );
+          }) ?? (
+            <StyledDefaultContent>
+              <Typography variant="body1">
+                {t(
+                  'Please select a step from the left panel to view its content.',
+                )}
+              </Typography>
+            </StyledDefaultContent>
+          )}
+          <CategoryContainer>
+            <ContinueButton onClick={() => {}} />
```

Comment:
The continue button doesn't work so it's hard to know if the logic in the useContext will work

Reply:
If you want to test the right panel logic, click on Ministry Expenses, then click on Mileage. I apologize for not making test instructions for this PR.

Reply:
I also added the handler for the continue button.